### PR TITLE
Make visualize more consistent with compute.

### PR DESCRIPTION
**Pull request by JSKenyon**
- [X] Closes #xxxx (N/A)
- [X] Tests added / passed
- [X] Passes `pre-commit run --all-files`

I have regularly encountered weird output from `visualize` when passing in Python collections/Xarray datasets. Consider the following example:

```python
import xarray
import dask.array as da
import dask


if __name__ == "__main__":

    xds = xarray.Dataset(
        {
            "DUMMY": (("x", "y"), da.ones((4, 4), name="ones"))
        }
    )

    dask.visualize(xds,
                   filename='graph.png',
                   optimize_graph=False)
```

This will produce the following output:
graph

Note the appearance of a strange seemingly disconnected node. In this trivial example, it doesn't muddy the water but these orphans can become highly connected in complicated graphs. This happens because an `xarray.Dataset` is simultaneously a `Mapping` and a dask collection and results in the `visualize` output becoming very difficult to interpret. The very minor change in this PR aims to make the behaviour of `visualize` more consistent with `compute` i.e. the graph produced by `visualize` should match the graph used in an identical `compute` call. 

With the changes in this PR, the output of the above is:
graph

Currently, this change breaks a single test: `dask/tests/test_base.py::test_visualize_lists`. The reason for this is that the test calls `visualize` on a `list` containing a `dict` describing a low-level graph. I can see why this may be useful, but this is [at odds with the docs](https://docs.dask.org/en/stable/graphviz.html) which state that `visualize` should be consistent with `compute`. I would argue that because compute cannot be called on the contents of this test, visualize should not be expected to work. Additionally, the current `visualize` code can be misleading as it doesn't properly traverse nested Python collections i.e. the results of `visualize` may not include all the input dask collections.

I admit my solution may be a bit heavy handed so I would appreciate other/better ideas!

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by quasiben** [comment]
add to allowlist

**Comment by jsignell** [comment]
I agree with your assessment about that test not testing functionality that we intend to support.

**Comment by JSKenyon** [comment]
Thanks @jsignell. Do you know what the procedure is in a case like this? I am not entirely sure who to ping for a review and whether or not I am allowed to remove the failing test. 

I do feel like this change will make the behaviour of visualise more reliable/intuitive.

**Comment by jsignell** [comment]
You should go ahead and remove the failing test. I will review and we can ping the rest of the maintenance team when we are both happy with it.

**Comment by JSKenyon** [comment]
I have removed the failing test and everything seems to be passing now.

File: dask/base.py [hunk 1]
```diff
@@ -630,25 +630,11 @@ def visualize(*args, **kwargs):
 
     filename = kwargs.pop("filename", "mydask")
     optimize_graph = kwargs.pop("optimize_graph", False)
+    traverse = kwargs.pop("traverse", True)
```
Comment (jsignell) [inline dask/base.py:633]: Can you add `traverse` to the docstring above? While you are at it you could probably also copy some of the docstring text from `compute.args` to make that section more explicit.

**Review by jsignell — APPROVED** [review]
Thanks for opening this @JSKenyon I think it's an improvement. I'll merge this next week unless anyone has objections (ping  @dask/maintenance)

File: dask/base.py (hunk 1 above)
Comment (JSKenyon) [reply dask/base.py:642]: Good point! Done!

**Comment by GenevieveBuckley** [comment]
No objections from me. 
I haven't played around with it, but the changes all look relatively sensible.

**Comment by jsignell** [comment]
I forgot to merge this, so I'm merging in main and will merge this today. Sorry for the delay @JSKenyon

**Comment by JSKenyon** [comment]
Thanks @jsignell. No worries about the delay. :-)

**Pull request merged**